Code/Framework/AzToolsFramework/AzToolsFramework/UI/DPEDebugViewer/DPEDebugTextView.cpp

```diff
@@ -0,0 +1,279 @@
+            }
+            Result RawKey(AZStd::string_view key, AZ::Dom::Lifetime)
+            {
+                NodeEntry* entry = CurrentNode();
+                if (entry != nullptr)
+                {
+                    entry->m_key = true;
+                }
+                Write(QStringLiteral(" "));
+                Write(QString::fromUtf8(key.data(), aznumeric_cast<int>(key.size())));
+                Write(QStringLiteral("="));
+                return VisitorSuccess();
+            }
+
+            Result StartArray()
+            {
+                StartValue();
+                Write(QStringLiteral("["));
+                m_stack.emplace();
+                m_stack.top().m_separator = QStringLiteral(", ");
+                return VisitorSuccess();
+            }
+            Result EndArray(AZ::u64)
+            {
+                m_stack.pop();
+                Write(QStringLiteral("]"));
+                return VisitorSuccess();
+            }
+
+            Result StartNode(AZ::Name name)
+            {
+                return RawStartNode(name.GetStringView(), AZ::Dom::Lifetime::Temporary);
+            }
+            Result RawStartNode(AZStd::string_view name, AZ::Dom::Lifetime)
+            {
+                int indentLevel = GetIndent();
+                if (auto currentNode = CurrentNode(); currentNode != nullptr && !currentNode->m_firstValue && !currentNode->m_nodeOpen)
+                {
+                    StartValue();
+                    Write(QStringLiteral("\n"));
+                    WriteIndent(indentLevel);
+                }
+                else
+                {
+                    StartValue();
+                }
+                Write(QStringLiteral("<"));
+                QString nodeName = QString::fromUtf8(name.data(), aznumeric_cast<int>(name.size()));
+                Write(nodeName);
+                m_stack.emplace();
+                m_stack.top().m_nodeOpen = true;
+                m_stack.top().m_nodeName = nodeName;
+                m_stack.top().m_indent = indentLevel + 1;
+                return VisitorSuccess();
+            }
+            Result EndNode(AZ::u64, AZ::u64)
+            {
+                if (m_stack.top().m_nodeOpen)
+                {
+                    Write(QStringLiteral(" />"));
+                }
+                else
+                {
+                    Write(QStringLiteral("\n"));
+                    WriteIndent(GetIndent() - 1);
+                    Write(QStringLiteral("</"));
+                    Write(m_stack.top().m_nodeName);
+                    Write(QStringLiteral(">"));
+                }
+                m_stack.pop();
+                return VisitorSuccess();
+            }
+
+        private:
+            struct NodeEntry
+            {
+                bool m_key = false;
+                bool m_nodeOpen = false;
+                bool m_firstValue = true;
+                QString m_separator = "";
+                QString m_nodeName;
```

Comment:
Pessimization here: It is quicker to default construct an empty string, than to use the `const char*` to construct a string as a `strlen` operation still needs to be performed to determine string is empty.

```suggestion
                QString m_separator;
                QString m_nodeName;
```

Reply:
I wanted to keep this explicit for clarity purposes so updated with `QStringLiteral("")` (which should also be free at runtime just copying the static string data ref)